perfdash/google-gcs-downloader.go

```diff
@@ -120,33 +115,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
 		for categoryLabel, categoryMap := range tests.Descriptions {
 			for testLabel, testDescriptions := range categoryMap {
 				for _, testDescription := range testDescriptions {
-					fileStem := fmt.Sprintf("artifacts/%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
-					artifacts, err := g.GoogleGCSBucketUtils.listFilesInBuild(job, buildNumber, fileStem)
+					filePrefix := fmt.Sprintf("%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
+					searchPrefix := fmt.Sprintf("artifacts/%v", filePrefix)
+					artifacts, err := g.GoogleGCSBucketUtils.listFilesInBuild(job, buildNumber, searchPrefix)
 					if err != nil || len(artifacts) == 0 {
-						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", fileStem, job, buildNumber, err)
+						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", searchPrefix, job, buildNumber, err)
 						continue
 					}
-					metricsFilename := artifacts[0][strings.LastIndex(artifacts[0], "/")+1:]
-					if len(artifacts) > 1 {
-						fmt.Printf("WARNING: found multiple %s files with data, reading only one: %s\n", fileStem, metricsFilename)
+					for _, artifact := range artifacts {
+						metricsFileName := filepath.Base(artifact)
+						resultCategory := getResultCategory(metricsFileName, filePrefix, categoryLabel, artifacts)
+						testDataResponse, err := g.GoogleGCSBucketUtils.getFileFromJenkinsGoogleBucket(job, buildNumber,
+							fmt.Sprintf("artifacts/%v", metricsFileName))
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "Error when reading response Body: %v\n", err)
+							continue
+						}
+						buildData := getBuildData(result, tests.Prefix, resultCategory, testLabel, job, resultLock)
```

Comment:
Don't you need to hold resultLock here?

Reply:
We actually hold the result lock when accessing the map, see the `getBuildData` impl

Reply:
You are right. It's OK.